src/request.jl

```diff
@@ -32,27 +32,70 @@ end
 @inline _remaining(x::Tuple) = Base.tail(x)
 @inline _remaining(@nospecialize(x)) = nothing
 @inline _first(::Tuple{}) = nothing
-@inline _first(x::Tuple{X,Vararg}) where {X<:Unsigned} = getfield(x,1)
+@inline _first(x::Tuple{X,Vararg}) where {X<:Unsigned} = getfield(x, 1)
 @inline _first(x::Union{Unsigned,Nothing}) = x
-@inline function _request_threads(num_requested::UInt32, wp::Ptr, ::StaticInt{N}, threadmask) where {N}
+@inline function _request_threads(
+  num_requested::UInt32,
+  wp::Ptr,
+  ::StaticInt{N},
+  threadmask,
+) where {N}
   ui, ft, num_requested, wp = __request_threads(num_requested, wp, _first(threadmask))
-  uit, ftt = _request_threads(num_requested, wp, StaticInt{N}()-StaticInt{1}(), _remaining(threadmask))
+  uit, ftt = _request_threads(
+    num_requested,
+    wp,
+    StaticInt{N}() - StaticInt{1}(),
+    _remaining(threadmask),
+  )
   (ui, uit...), (ft, ftt...)
 end
-@inline function _request_threads(num_requested::UInt32, wp::Ptr, ::StaticInt{1}, threadmask)
+@inline function _request_threads(
+  num_requested::UInt32,
+  wp::Ptr,
+  ::StaticInt{1},
+  threadmask,
+)
   ui, ft, num_requested, wp = __request_threads(num_requested, wp, _first(threadmask))
-  (ui, ), (ft, )
+  (ui,), (ft,)
 end
+
+@inline function _request_threads_recurse(
+  ::Tuple{},
+  num_requested::UInt32,
+  wp::Ptr,
+  threadmask,
+  ::Int,
+)
+  _request_threads(num_requested, wp, StaticInt{1}(), threadmask)
+end
+@inline function _request_threads_recurse(
+  tup::Tuple{StaticInt{N},Vararg},
+  num_requested::UInt32,
+  wp::Ptr,
+  threadmask,
+  i::Int,
+) where {N}
+  if i == N
+    _request_threads(num_requested, wp, StaticInt{N}(), threadmask)
+  else
+    _request_threads_recurse(Base.tail(tup), num_requested, wp, threadmask, i)
+  end
+end
+
+
 @inline function _request_threads(num_requested::UInt32, wp::Ptr, i::Int, threadmask) # fallback in absence of static scheduling
```

Comment:
Is this fallback necessary anymore? The worker_bits are static again (moving back to the overprovisioned `CPUSummary.sys_threads` so this never gets used). I disabled it in this pull request and it seems to work fine: https://github.com/JuliaSIMD/PolyesterWeave.jl/pull/13